maintenance/measure_tile_performance/measure_tile_performance.py

```diff
@@ -15,24 +15,52 @@
 NAME = 'snowplow'
 HOST = 'redshift.analytics.gov.bc.ca'
 PORT = 5439
+USER = os.environ['lookeruser_rs']
+PASSWD = os.environ['lookerpass_rs']
+CONNECTION_STRING = (
+    f"dbname='{NAME}' "
+    f"host='{HOST}' "
+    f"port='{PORT}' "
+    f"user='{USER}' "
+    f"password={PASSWD}")
 
 # connects with redshift database using environment variables
 # set redshift cache to 'off' before running any looker API calls
 # run the API calls
 # set redshift cache back to 'on' after the queries are processed.
 # close the redshift connection
+
+def cache(status):
+ # set redshift cache back to 'on' or 'off'
+    sleep_timer = 0
+    while True and sleep_timer <= 50: 
+        conn = psycopg2.connect(dsn=CONNECTION_STRING)
+        with conn:
+            with conn.cursor() as curs:
+                try:
+                    curs.execute(f'ALTER USER looker SET enable_result_cache_for_session TO {status} ;')
+                    print(f'redshift cache is {status} and connection is closed')
+                except Exception as err:
+                    if status == "on":
+                        #  linear backoff loop to reconnect with database 5 times
+                        sleep_timer += 10
+                        if sleep_timer > 50:
+                            print(f'URGENT! After retrying redshift connection 5 times, program is exiting due to psycopg2 execution error: {err}')
+                            print('Requires manual run of: ALTER USER looker SET enable_result_cache_for_session TO on;')
+                            exit(1)
+                        else:    
+                            print(f"Retrying connection after {sleep_timer} seconds")
+                            time.sleep(sleep_timer)
+                            continue
```

Comment:
I think you want to have another `conn.close()` call into the else block of the Exception above (before the `continue` is reached), in order to close it prior to de-referencing the connection from the conn variable in line 37 at the start of the loop; otherwise you have a accumulate connection objects each time you repeat the loop.

Reply:
Closed the connection in the latest commit, please re-do the testing and let me know if it can be merged.